data_structures/ex_1/binary_search_tree.py

```diff
@@ -5,6 +5,30 @@ def __init__(self, value):
     self.right = None
 
   def depth_first_for_each(self, cb):
+    search_list = []
```

Comment:
It seems like you're trying to implement BFS here instead of DFS. DFS is rather straight forward. Here's some pseudo-code:

```
func dfs(cb):
  cb(self.value)

  if left:
    recursively call dfs on self.left
  if right:
    recursively call dfs on self.right
```